packages/camera/camera_android_camerax/test/android_camera_camerax_test.dart

```diff
@@ -311,46 +421,93 @@ void main() {
         camera.processCameraProvider!.unbind(<UseCase>[camera.preview!]));
   });
 
-  test('resumePreview does not bind preview to lifecycle if already bound',
+  test(
+      'resumePreview does not bind preview to lifecycle or update camera state observers if already bound',
       () async {
     final AndroidCameraCameraX camera = AndroidCameraCameraX();
+    final MockProcessCameraProvider mockProcessCameraProvider =
+        MockProcessCameraProvider();
+    final MockCamera mockCamera = MockCamera();
+    final MockCameraInfo mockCameraInfo = MockCameraInfo();
+    final MockLiveData mockLiveCameraState = MockLiveData();
 
-    camera.processCameraProvider = MockProcessCameraProvider();
+    camera.processCameraProvider = mockProcessCameraProvider;
     camera.cameraSelector = MockCameraSelector();
     camera.preview = MockPreview();
 
     when(camera.processCameraProvider!.isBound(camera.preview!))
         .thenAnswer((_) async => true);
 
+    when(mockProcessCameraProvider
+            .bindToLifecycle(camera.cameraSelector, <UseCase>[camera.preview!]))
+        .thenAnswer((_) async => mockCamera);
+    when(mockCamera.getCameraInfo()).thenAnswer((_) async => mockCameraInfo);
+    when(mockCameraInfo.getLiveCameraState())
+        .thenAnswer((_) async => mockLiveCameraState);
+
     await camera.resumePreview(78);
 
     verifyNever(camera.processCameraProvider!
         .bindToLifecycle(camera.cameraSelector!, <UseCase>[camera.preview!]));
+    verifyNever(mockLiveCameraState.observe(any));
   });
 
-  test('resumePreview binds preview to lifecycle if not already bound',
+  test(
+      'resumePreview binds preview to lifecycle and updates camera state observers if not already bound',
       () async {
     final AndroidCameraCameraX camera = AndroidCameraCameraX();
+    final MockProcessCameraProvider mockProcessCameraProvider =
```

Comment:
A bunch of these tests need the same default mock values. Consider putting them in a setup method to make these tests easier to follow.